src/state.rs

```diff
@@ -219,24 +220,22 @@ impl<T: StateData> Stage for StateTransitionStage<T> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, StageLabel)]
+pub struct StateTransitionStageLabel(TypeId, String);
```

Comment:
This is to make sure that the stage label is compatible between the two extension traits. If you like, I could add a `cfg` attribute here checking for at least one of `app` or `schedule` flags.